src/Domain/LeanCode.DomainModels.MassTransitRelay/MassTransitRelayServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,38 @@
+using LeanCode.DomainModels.MassTransitRelay.Testing;
+using MassTransit;
+using MassTransit.Testing.Implementations;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace LeanCode.DomainModels.MassTransitRelay;
+
+public static class MassTransitRelayServiceCollectionExtensions
+{
+    public static void AddCQRSMassTransitIntegration(
+        this IServiceCollection services,
+        Action<MassTransitHostOptions>? hostCfg = null
+    )
+    {
+        var interceptor = new AsyncEventsInterceptor();
+        interceptor.Configure();
+        services.AddSingleton(interceptor);
+
+        hostCfg ??= opts =>
```

Comment:
This should be controlled by the parent IMO, we shouldn't force that.

Reply:
I feel that lack of it will be a painpoint in our projects - we will forget about it. 
Combined with what I've said in the description maybe we need a split ->

`serviceCollection.AddAsyncEventsInterceptor()` - self explanatory

`serviceColllection.AddCQRSMassTransitIntegration(Action<MassTransitHostOptions>, Action<IBusRegistrationConfigurator>)` - this one will call the first one + register MT and will be intended for regular use.

But if sb needs other way to register mass transit, e.g. through test harness, they'll have a possibility.

Reply:
There is nothing being forced here as far as I can tell, unless @jakubfijalkowski means we shouldn't `AddOptions<MassTransitHostOptions>()` here. But having `opts.WaitUntilStarted = true;` sounds like a reasonable default and callers can always opt out of it by passing their own `hostCfg`.

Reply:
> I feel that lack of it will be a painpoint in our projects - we will forget about it.

I agree, we will forget that, but this is no excuse to break SRP - you're saying that you configure the relay and force user to configure MT separately, yet you are configuring MT to some extent.

> There is nothing being forced here as far as I can tell

You are forced to opt out of it (which it seemingly does not do), which is not intuitive.

Reply:
> which is not intuitive

You're not incorrect, but it's still something we should probably encourage by default. If not here, maybe in the template?

Reply:
I'm only arguing that it should not be forced here. We might have some other way of ensuring that (even some `AddDefaultMTConfiguration`) or put that in template, but not _here_.

Reply:
:popcorn: 

So, as I said maybe sth like this:
```
    public static IServiceCollection AddCQRSMassTransitIntegration(this IServiceCollection services,
        Action<IBusRegistrationConfigurator> busCfg,
        Action<MassTransitHostOptions>? hostCfg = null)
    {
        services.AddAsyncEventsInterceptor();

        hostCfg ??= opts =>
        {
            opts.WaitUntilStarted = true;
        };

        services.AddOptions<MassTransitHostOptions>().Configure(hostCfg);
        services.AddMassTransit(busCfg);

        return services;
    }

    public static IServiceCollection AddAsyncEventsInterceptor(
        this IServiceCollection services)
    {
        var interceptor = new AsyncEventsInterceptor();
        interceptor.Configure();
        services.AddSingleton(interceptor);

        return services;
    }
```